src/composables/useCamera/index.ts

```diff
@@ -1,42 +1,39 @@
-import { computed, watchEffect, onUnmounted, ref } from 'vue'
 import { Camera, OrthographicCamera, PerspectiveCamera } from 'three'
+import { computed, ref, triggerRef, watchEffect } from 'vue'
 
-import type { TresScene } from '../../types'
 import type { TresContext } from '../useTresContextProvider'
 
 
-export const useCamera = ({ sizes, scene }: Pick<TresContext, 'sizes'> & { scene: TresScene }) => {
-
-  // the computed does not trigger, when for example the camera postion changes
-  const cameras = ref<Camera[]>([])
+export const useCamera = ({ sizes, scene }: Pick<TresContext, 'sizes' | 'scene'>) => {
+  const cameras = computed<Camera[]>(() => scene.value.children.filter((i): i is Camera => (i as any).isCamera))
+  const activeCameraUuid = ref(cameras.value?.[0]?.uuid ?? "")
   const camera = computed<Camera | undefined>(
-    () => cameras.value[0]
+    () => cameras.value.find(c => c.uuid === activeCameraUuid.value) ?? cameras.value.at(0),
   )
 
   const addCamera = (newCamera: Camera, active = false) => {
     if (cameras.value.some(({ uuid }) => uuid === newCamera.uuid))
       return
 
+    scene.value.add(newCamera)
```

Comment:
I think nothing in this composable should be responsible for adding things to the scene tree. That logic already exists here:
https://github.com/Tresjs/tres/blob/81b7914fc685aa7314541480a96801c0cc7fd542/src/core/nodeOps.ts#L111

Can you explain, why you added this code? Was something not working correctly?

Reply:
> I did some refactoring and realized the camera hasn't been truly added so I add this line
I don't think this is true. The camera ends up in the scene and I'm pretty sure it is because of this line:
https://github.com/Tresjs/tres/blob/81b7914fc685aa7314541480a96801c0cc7fd542/src/core/nodeOps.ts#L111
Could you maybe provide some code that proofs your statemen?

With the solution you proposed, cameras are added at the root level of the scene. So the component hierarchy is not respected anymore. The newly added code would prevent someone from adding a camera to a TresGroup for example.

Reply:
if add a camera by adding `<TresPerspectiveCamera/>` it goes here that's right, but if using `addCamera` which in this composable will never add to the scene(like addDefaultCamera), it works just because in the render use the camera to render the scene in `onLoop`.

simply, you can remove the code here

https://github.com/Tresjs/tres/blob/81b7914fc685aa7314541480a96801c0cc7fd542/src/composables/useRenderer/index.ts#L238

then create a scene without `<TresPerspectiveCamera/>` in TresCanvas, see we have no default camera has been added, just a warning and blank scene

> [TresJS ▲ ■ ●]  No camera found. Creating a default perspective camera. To have full control over a camera, please add one to the scene.

Reply:
yeah, the same camera would be added twice, that's my mistake, things about `tres__registerCamera` and `tres__deregisterCamera` should be removed after this